### PR TITLE
Fix deprecation warning from Tomli

### PR DESCRIPTION
**Pull request by hukkin**
- Pass a binary file object to Tomli to fix a deprecation warning
- ~~Turn DeprecationWarnings in the test suite to errors~~

### CODE REVIEW

File: src/towncrier/_settings.py [hunk 1]
```diff
@@ -74,9 +75,12 @@ def load_config(directory):
...
 
 def load_config_from_file(directory, config_file):
-
-    with io.open(config_file, "r", encoding="utf8") as conffile:
-        config = _toml_load(conffile)
+    if tomli:
+        with open(config_file, "rb") as conffile:
+            config = tomli.load(conffile)
+    else:
+        with io.open(config_file, "r", encoding="utf8") as conffile:
```
Comment (graingert) [inline src/towncrier/_settings.py:82]:
```suggestion
        with io.open(config_file, "r", encoding="utf8", newlines="") as conffile:
```

File: src/towncrier/_settings.py [hunk 2]
```diff
@@ -74,9 +75,12 @@ def load_config(directory):
 
 
 def load_config_from_file(directory, config_file):
-
-    with io.open(config_file, "r", encoding="utf8") as conffile:
-        config = _toml_load(conffile)
+    if tomli:
+        with open(config_file, "rb") as conffile:
```
Comment (graingert) [inline src/towncrier/_settings.py:79]:
```suggestion
        with io.open(config_file, "rb") as conffile:
```

File: src/towncrier/_settings.py (hunk 1 above)
Comment (hukkin) [reply src/towncrier/_settings.py]: Thanks!

File: src/towncrier/_settings.py (hunk 2 above)
Comment (hukkin) [reply src/towncrier/_settings.py:79]: This should be unnecessary in Python 3.6+ (which is always true in this branch)?

**Comment by graingert** [comment]
hmm, does this need something to enable deprecation warnings in tests?

File: src/towncrier/_settings.py (hunk 2 above)
Comment (graingert) [reply src/towncrier/_settings.py:79]: yeah it's unnecessary, but I'm going to pyupgrade all these away in one go

File: src/towncrier/_settings.py [hunk 3]
```diff
@@ -74,9 +75,12 @@ def load_config(directory):
...
+            config = tomli.load(conffile)
+    else:
+        with io.open(config_file, "r", encoding="utf8", newlines="") as conffile:
```
Comment (hukkin) [inline src/towncrier/_settings.py:82]:
```suggestion
        with io.open(config_file, "r", encoding="utf8", newline="") as conffile:
```

**Comment by graingert** [comment]
oops

**Comment by hukkin** [comment]
> hmm, does this need something to enable deprecation warnings as failures in tests?

This doesnt seem to use a use a widely used test runner so I'm a bit clueless how to implement this :smile:

File: src/towncrier/_settings.py (hunk 2 above)
Comment (hukkin) [reply src/towncrier/_settings.py:79]: Haha, ok ill commit this if you want more work for pyupgrade? :smile:

**Comment by graingert** [comment]
> This doesnt seem to use a use a widely used test runner 
😬

**Comment by hukkin** [comment]
Ah it's twisted's own test runner :smile: Sorry for my ignorance, didn't mean to undermine it. What I meant to say is I personally have ever done this in pytest :smile: 

I think prefixing the test runner invocation in `tox.ini` with `python -W error::DeprecationWarning` should probably do the trick. Should I add it?

**Comment by hukkin** [comment]
I tested that the `python -W error::DeprecationWarning -m` addition actually makes the test suite fail in master branch and it does. So commited that change here too.

**Comment by graingert** [comment]
https://github.com/twisted/towncrier/runs/3376146214#step:7:60 ah that's an upstream twisted bug.
ok best leave it off for now and we'll have to do it later

**Comment by hukkin** [comment]
:+1: reverted that change

**Comment by graingert** [comment]
this doesn't need a newsfragment because tomli support hasn't been released yet

**Review by graingert — APPROVED** [review]

**Comment by adiroiban** [comment]
I guess that this can be merged :)

**Pull request merged**